Sources/Hyperspace/Service/Network/NetworkActivityIndicatable.swift

```diff
@@ -10,7 +10,11 @@ import Foundation
 
 /// Represents an object that can indicate to the user that network activity is currently taking place.
 public protocol NetworkActivityIndicatable {
+    #if MACOS
+    //macOS does not have a activity indicator
```

Comment:
This doesn't need to be here - the point of abstracting UIApplication's functionality into a protocol is so a custom indicator could be built on any platform.

In the case of macOS, if the client doesn't want to create a custom indicator they can just pass nil into the NetworkService.

Having said that, we will probably need to remove the default value of UIApplication.shared.

Reply:
As for removing the default value of `UIApplication.shared`, do you think we could use the #if macros to define two different function signatures? One that sets no default for macOS and then one that sets the default for everything else?

Reply:
We could either do that or create an extension on NetworkService that is only compiled into iOS/tvOS. The base initializer would have no default, the iOS/tvOS extension would have the default.


Names may need to be tweaked to avoid ambiguity, but I think that's a little cleaner than #if, because now that I think about it watchOS doesn't have UIApplication either.

Reply:
Yeah, that sounds like a good approach.